File: leetcode1282/solution.py
class Solution:
    def groupThePeople(self, groupSizes: List[int]) -> List[List[int]]:
        ans = []
        ht = {}

        for i in range(len(groupSizes)):
            if groupSizes[i] not in ht:
                ht[groupSizes[i]] = [i]
            else:
                ht[groupSizes[i]].insert(0, i)
            if len(ht[groupSizes[i]]) == groupSizes[i]:
                ans.insert(0, ht[groupSizes[i]])
                ht[groupSizes[i]] = []

        return ans

File: leetcode1282/solution.rs
use std::collections::HashMap;

impl Solution {
    pub fn group_the_people(group_sizes: Vec<i32>) -> Vec<Vec<i32>> {
        let mut ht: HashMap<i32, Vec<i32>> = HashMap::new();

        let mut ans: Vec<Vec<i32>> = vec![];

        for (idx, size) in group_sizes.iter().enumerate() {
            
            ht.entry(*size).or_insert(Vec::new()).push(idx as i32);

            if let Some(v) = ht.get_mut(size) {
                if v.len() as i32 == *size {
                    ans.push(v.to_vec());
                    v.clear();
                }
            }


        }

        ans
    }
}